io-actions/src/fragment/replace.js

```diff
@@ -0,0 +1,105 @@
+const fetch = require('node-fetch');
+const { odinReferences, odinPath } = require('./paths.js');
+const DICTIONARY_ID_PATH = 'dictionary/index';
+const PH_REGEXP = /{{(\s*([\w\-]+)\s*)}}/gi;
+
+const getDictionaryId = async ({ surface, locale }) => {
+    try {
+        const dictionaryPath = odinPath(surface, locale, DICTIONARY_ID_PATH);
+        const response = await fetch(dictionaryPath);
+        if (response.status == 200) {
+            const { items } = await response.json();
+            if (items?.length == 1) {
+                return items[0].id;
+            }
+        }
+    } catch (e) {
+        console.error(e);
+    }
+    return null;
+};
+
+const extractValue = (ref) => {
+    const value = ref.value || ref?.richTextValue?.value || '';
+    // Escape control characters before parsing
+    return value.replace(/[\u0000-\u001F\u007F-\u009F]/g, '');
+};
+
+const getDictionary = async (context) => {
```

Comment:
I recommend to use function instead of arrow functions like `const getDictionary`, the reason is that, afaik in case of error, the function name is not logged in the console, but rather logged as an anonymous function.